CONTRIBUTING.md

```diff
@@ -0,0 +1,132 @@
+
+```bash
+cargo check --all
+```
+
+`nextest` is necessary for `make test` & `make unit-test` to work. Install the crate with:
+
+```bash
+cargo install cargo-nextest --locked
+```
+
+You can run the test suite alone, or just run a specific test:
+
+```bash
+# Run the full suite
+make test
+```
+
+`tracing-active-tree` follows the Rust community coding style. We use Rustfmt and [Clippy](https://github.com/Manishearth/rust-clippy) to automatically format and lint our code. Using these tools is checked in our CI. These are as part of `make check`.
+
+```bash
+make check
+```
+
+### Finding something to work on
+
+For beginners, we have prepared many suitable tasks for you. Checkout our [Help Wanted issues](https://github.com/tikv/tracing-active-tree/issues?q=is%3Aopen+is%3Aissue+label%3Astatus%2Fhelp-wanted) for a list, in which we have also marked the difficulty level.
+
+If you are planning something big, for example, relates to multiple components or changes current behaviors, make sure to open an issue to discuss with us before going on.
+
+### Linking issues
+
+Code repositories in `tracing-active-tree` community require **ALL** the pull requests referring to its corresponding issues. In the pull request body, there **MUST** be one line starting with `Issue Number:` and linking the relevant issues via the [keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword), for example:
+
+If the pull request resolves the relevant issues, and you want GitHub to close these issues automatically after it merged into the default branch, you can use the syntax (`KEYWORD #ISSUE-NUMBER`) like this:
+
+```issue-message
+Issue Number: close #123
+```
+
+If the pull request links an issue but does not close it, you can use the keyword `ref` like this:
+
+```issue-message
+Issue Number: ref #456
+```
+
+Multiple issues should use full syntax for each issue and separate by a comma, like:
+
+```issue-message
+Issue Number: close #123, ref #456
+```
+
+For pull requests trying to close issues in a different repository, contributors need to first create an issue in the same repository and use this issue to track.
+
+If the pull request body does not provide the required content, the bot will add the `do-not-merge/needs-linked-issue` label to the pull request to prevent it from being merged.
+
+### Format of the commit message
+
+The bot we use will extract the pull request title as the one-line subject and messages inside the `commit-message` code block as commit message body. For example, a pull request with title `pkg: what's changed in this one package` and body containing:
+
+    ```commit-message
+    any multiple line commit messages that go into
+    the final commit message body
+
+    * fix something 1
+    * fix something 2
+    ```
```

Comment:
Is it a GitHub feature? It seems we don't have such infrastructure...

Reply:
I think yes. How about just link to [this](https://github.com/pingcap/community/blob/master/contributors/commit-message-pr-style.md) ?

Reply:
linked it.